app/models/primer.rb

```diff
@@ -1,9 +1,9 @@
 class Primer < ActiveRecord::Base
 
   has_many :marker_assays_A, class_name: 'MarkerAssay',
```

Comment:
That's because `marker_assays` links to `primers` twice (via `primer_a_id` and `primer_b_id`). I've added a `marker_assays` method to `Primer` which pulls a union of both lists.